### PR TITLE
Pull request for issue #140, team 1

### PR DESCRIPTION
**Pull request by NejcZupec**
Now list of online users is initially displayed.

### CODE REVIEW

File: piplmesh/frontend/context_processors.py [hunk 1]
```diff
@@ -10,6 +10,7 @@ def global_vars(request):
 
     return {
         'HOME_CHANNEL_ID': views.HOME_CHANNEL_ID,
+        'LOGGED_IN_USERS': views.get_logged_in_users(),
```
Comment (mitar) [inline piplmesh/frontend/context_processors.py]: No, this should go only to context of home view.

Comment (mitar) [inline piplmesh/frontend/context_processors.py]: Furthermore, it is not a constant. So it should not be an all upper-case variable name.

File: piplmesh/frontend/templates/userlist.html [hunk 2]
```diff
@@ -1,3 +1,7 @@
 <div>
-    <ul id="userlist"></ul>
+    <ul id="userlist">
+        {% for user in LOGGED_IN_USERS %}
+            <li>{{user}}</li>
```
Comment (mitar) [inline piplmesh/frontend/templates/userlist.html]: Spaces inside the `{{ }}` tag.

File: piplmesh/account/views.py [hunk 3]
```diff
@@ -105,3 +105,9 @@ def process_channel_unsubscribe(sender, request, channel_id, **kwargs):
         pull__connections=None,
         set__connection_last_unsubscribe=datetime.datetime.now(),
     )
+
+def get_logged_in_users():
+    users = models.User.objects(
```
Comment (mitar) [inline piplmesh/account/views.py]: There is probably no need to split this into multiple lines.

File: piplmesh/account/views.py [hunk 4]
```diff
@@ -105,3 +105,9 @@ def process_channel_unsubscribe(sender, request, channel_id, **kwargs):
...
+        is_online = True
```
Comment (mitar) [inline piplmesh/account/views.py]: But if you did it, you should put a comma at the end.

File: piplmesh/frontend/templates/userlist.html (hunk 2 above)
Comment (NejcZupec) [inline piplmesh/frontend/templates/userlist.html]: Done.

File: piplmesh/account/views.py (hunk 4 above)
Comment (NejcZupec) [inline piplmesh/account/views.py]: Done.

**Comment by mitar** [comment]
Pull request not yet updated?

**Comment by NejcZupec** [comment]
Not yet. It'll be soon.

File: piplmesh/frontend/context_processors.py (hunk 1 above)
Comment (NejcZupec) [inline piplmesh/frontend/context_processors.py]: I removed this line of code.

**Comment by NejcZupec** [comment]
This version is not working. I'm not sure how to resolve this. Is this the right way ?

File: piplmesh/frontend/templates/userlist.html [hunk 5]
```diff
@@ -1,3 +1,7 @@
 <div>
-    <ul id="userlist"></ul>
+    <ul id="userlist">
+        {% for user in logged_in_users %}
```
Comment (mitar) [inline piplmesh/frontend/templates/userlist.html]: Hm, probably better is _online_users_.

File: piplmesh/frontend/views.py [hunk 6]
```diff
@@ -1,7 +1,16 @@
+from django import shortcuts
 from django.views import generic as generic_views
 
+from piplmesh.account import models
+
 class HomeView(generic_views.TemplateView):
     template_name = 'home.html'
+    
+    def get_logged_in_users(self, request):
```
Comment (mitar) [inline piplmesh/frontend/views.py]: What is this? Where did you get the idea to do it like this? Can you check documentation on class-based views and how context is passed to templates?

File: piplmesh/frontend/templates/userlist.html (hunk 5 above)
Comment (NejcZupec) [inline piplmesh/frontend/templates/userlist.html]: Ok.

File: piplmesh/frontend/views.py (hunk 6 above)
Comment (NejcZupec) [inline piplmesh/frontend/views.py]: Ok, now I'm reading this document: https://docs.djangoproject.com/en/dev/topics/class-based-views/ :P

Comment (mitar) [inline piplmesh/frontend/views.py]: Just now? Ccc.

(Yes, I noticed that you tried to take the shortcut. Thanks for spending me time pointing you to documentation.)

Comment (NejcZupec) [inline piplmesh/frontend/views.py]: Yes, thank you Mitar. I didn't know, where this code should be. Now I know :)

I've pushed new commit just now.

File: piplmesh/frontend/views.py [hunk 7]
```diff
@@ -1,7 +1,14 @@
...
 
+from piplmesh.account import models
+
 class HomeView(generic_views.TemplateView):
     template_name = 'home.html'
-        
+    
+    def get_context_data(self, **kwargs):
+        context = super(HomeView, self).get_context_data(**kwargs)
+        context['online_users'] = models.User.objects(is_online = True)
```
Comment (mitar) [inline piplmesh/frontend/views.py]: No spaces around `=`.

**Comment by mitar** [comment]
Have you tested the code? Does it work OK now?

**Comment by mitar** [comment]
So is user list working now as expected?

File: piplmesh/frontend/views.py (hunk 7 above)
Comment (NejcZupec) [inline piplmesh/frontend/views.py]: Done.

**Comment by NejcZupec** [comment]
Yes, I've tested code and it's working.

**Pull request merged**